### PR TITLE
Guard pruneExecutions against overflow and non-positive age deleting recent executions

### PR DESCRIPTION
**Pull request by nikhiln64**
Pull Request type
----
- [x] Bugfix
- [ ] Feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] WHOSUSING.md
- [ ] Other (please describe):

**NOTE**: Please remember to run `./gradlew spotlessApply` to fix any format violations.

Changes in this PR
----

`pruneExecutions` computed its cutoff as `Instant.now().minus(olderThanDays, DAYS).toEpochMilli()` with no bounds check. Two ways that turned the prune endpoint into a data-loss operation:

- very large values (the issue's `olderThanDays=99999`) push the computed epoch **negative**, and the search backend matched recent executions against the negative bound, which were then hard-deleted in batches
- non-positive values put the cutoff **in the future**, matching every terminal execution

The fix extracts the cutoff computation into `computePruneCutoffEpochMs(int, Instant)`: rejects `olderThanDays < 1` with a clear message, and clamps the cutoff to epoch start, where `endTime < 0` correctly matches nothing ("older than anything that exists").

Extracted as a static method with an injectable clock so the boundary behavior is unit-testable deterministically: `AgentServicePruneCutoffTest` covers the normal case, the 99999-day clamp, the largest non-clamped age, and rejection of zero/negative. 4/4 green; `spotlessApply` run.

Issue #1331

Alternatives considered
----

Also considered chasing why the search backend matches positive `endTime` values against a negative long bound (the second layer of the bug) - worth a follow-up in the search query serialization, but the guard here closes the data-loss hole regardless of backend behavior.

Developed with AI assistance (Claude Code), reviewed and driven by a human.

### CODE REVIEW

**Review by v1r3n — COMMENTED** [review]
Looks good. The cutoff helper rejects unsafe non-positive ages, clamps pre-epoch cutoffs, is wired into the actual prune path, and has deterministic coverage for the boundary cases. The existing controller advice maps the new `IllegalArgumentException` to a `400 Bad Request`, so invalid API input is handled correctly.

File: agentspan/src/main/java/org/conductoross/conductor/ai/agentspan/runtime/service/AgentService.java [hunk 1]
```diff
@@ -576,8 +576,28 @@ public void deleteExecutionRecord(String executionId, boolean archiveTasks) {
      * @param archiveTasks if true, archive task records instead of deleting
      * @return number of executions deleted
      */
```
Comment (kowser-orkes) [inline agentspan/src/main/java/org/conductoross/conductor/ai/agentspan/runtime/service/AgentService.java]: This documentation is for `pruneExecutions` is now dangling. Can you please fix it?

**Comment by nikhiln64** [comment]
The branch was just updated to the latest main and the workflow run is waiting for maintainer approval. Whenever someone has a moment to approve it, the PR is otherwise ready for review.

**Review by kowser-orkes — CHANGES_REQUESTED** [review]

File: agentspan/src/main/java/org/conductoross/conductor/ai/agentspan/runtime/service/AgentService.java (hunk 1 above)
Comment (kowser-orkes) [reply agentspan/src/main/java/org/conductoross/conductor/ai/agentspan/runtime/service/AgentService.java]: Thanks for the PR. It looks good to me except that documentation issue got created

Comment (nikhiln64) [reply agentspan/src/main/java/org/conductoross/conductor/ai/agentspan/runtime/service/AgentService.java]: Good catch, thank you. The helper insertion left the pruneExecutions javadoc stranded above it. Moved the doc back onto the method and pushed.

File: agentspan/src/main/java/org/conductoross/conductor/ai/agentspan/runtime/service/AgentService.java [hunk 2]
```diff
@@ -566,6 +566,26 @@ public void deleteExecutionRecord(String executionId, boolean archiveTasks) {
...
+     * future (matching every terminal execution), and very large values push the computed epoch
+     * negative, which the search backend matched against recent executions. A cutoff clamped to
+     * epoch start matches nothing, which is the correct meaning of "older than anything that
+     * exists".
+     *
+     * @param olderThanDays minimum age in days, must be >= 1
+     * @param now the current instant
+     * @return cutoff in epoch milliseconds, never negative
+     */
+    static long computePruneCutoffEpochMs(int olderThanDays, Instant now) {
```
Comment (kowser-orkes) [inline agentspan/src/main/java/org/conductoross/conductor/ai/agentspan/runtime/service/AgentService.java:583]: It should be `private`

**Review by kowser-orkes — CHANGES_REQUESTED** [review]
Can you please rebase and squash all changes. Three commits seems redundant :-)

File: agentspan/src/main/java/org/conductoross/conductor/ai/agentspan/runtime/service/AgentService.java (hunk 2 above)
Comment (nikhiln64) [reply agentspan/src/main/java/org/conductoross/conductor/ai/agentspan/runtime/service/AgentService.java:583]: I kept it package private on purpose so AgentServicePruneCutoffTest can exercise the boundary cases deterministically with a fixed clock, since making it fully private would force the tests to construct the whole service. I have annotated it with @VisibleForTesting following the pattern in WorkflowMonitor and WorkflowRepairService so the intent is explicit. If you would still rather have it private I am happy to switch and rework the tests through the public method instead, wdyt.

**Comment by nikhiln64** [comment]
Done, rebased onto the latest main and squashed everything into a single commit. On your earlier note about visibility, I annotated the helper with @VisibleForTesting following the WorkflowMonitor pattern since the boundary tests call it directly with a fixed clock, but if you would rather have it fully private I will rework the tests through the public method. The fresh CI run will need an approval when someone gets a chance.

**Review by kowser-orkes — APPROVED** [review]
